drivers/soundwire/cadence_master.c

```diff
@@ -222,6 +222,24 @@ static int cdns_clear_bit(struct sdw_cdns *cdns, int offset, u32 value)
 		if ((reg_read & value) == 0)
 			return 0;
 
+		timeout--;
+		usleep_range(50, 100);
+	} while (timeout != 0);
+
+	return -EAGAIN;
+}
+
+static int cdns_set_wait(struct sdw_cdns *cdns, int offset, u32 mask, u32 value)
+{
+	int timeout = 10;
+	u32 reg_read;
+
+	/* Wait for bit to be set */
+	do {
+		reg_read = readl(cdns->registers + offset);
+		if ((reg_read & mask) == value)
+			return 0;
+
```

Comment:
Will it be better to return -ETIMEDOUT on line 247 ?

Reply:
yes, I prefer -ETIMEDOUT